src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -742,27 +740,22 @@ private static String getProviderUrl(String server, String rootDN) {
         return server;
     }
 
-    /**
-     * Creates security components.
-     * @return Created {@link SecurityComponents}
-     * @throws IllegalStateException Execution error
-     */
     @Override @Nonnull
     public SecurityComponents createSecurityComponents() {
         if (configurations.size() > 1) {
```

Comment:
History lesson for context;
When we implemented the multiple ldap server support we were a bit paranoid of changing the behavior in any way for configurations with only one server configured. That's why you see all of these `if (configurations.size() > 1)` all over the place. It was also the reason for the `DelegatedLdapUserDetails`  class that I see you have changed to always be used instead of before; only when multiple configurations are in effect.

Since the behavior _(both user wise but also in binary compatibility)_ is changing now and `DelegatedLdapUserDetails` is used all the time, I believe it's now possible to remove all this semi redundant code that has separate treatments for the single server configuration scenario.

But I won't block this PR if you don't want to include that cleanup work here ;).

Reply:
Trying this in #52.